src/osd/osd_types.cc

```diff
@@ -1483,7 +1483,8 @@ void pg_pool_t::encode(bufferlist& bl, uint64_t features) const
   }
 
   uint8_t v = 24;
-  if (!(features & CEPH_FEATURE_NEW_OSDOP_ENCODING)) {
+  if (!(features & CEPH_FEATURE_NEW_OSDOP_ENCODING) ||
+      !(features & CEPH_FEATURE_SERVER_JEWEL)) {
```

Comment:
> but v0.94.9 does not have CEPH_FEATURE_NEW_OSDOP_ENCODING defined

During the upgrade the cluster is mixed: there are both 0.94.9 and 10.2.5. In particular if all monitors run jewel quorum_features contains CEPH_FEATURE_NEW_OSDOP_ENCODING (and CEPH_FEATURE_SERVER_JEWEL):
<pre>
2017-01-25 08:11:06.271957 7ffb87fff700 10 mon.saceph-mon1@0(electing).paxosservice(auth 1..14) restart
2017-01-25 08:11:06.272063 7ffb87fff700 10 mon.saceph-mon1@0(electing) e1 win_election epoch 16 quorum 0,2 features 576460752032874495
</pre>

OSDMonitor::encode_pending() understands that there are some hammer OSDs, and clears CEPH_FEATURE_SERVER_JEWEL to encode the map like hammer does (to get a correct full_crc). Therefore pg_pool_t::encode() should check for CEPH_SERVER_JEWEL, and use hammer compatible encoding (v21) if this bit is not set.